services/app-api/src/postgres/contractAndRates/insertRate.ts

```diff
@@ -59,28 +125,30 @@ async function insertDraftRate(
                 },
             })
 
-            return {
+            const finalRate: RateType = {
                 id: rate.id,
+                status: 'DRAFT',
+                stateCode: rate.stateCode,
+                stateNumber: rate.stateNumber,
                 revisions: rate.revisions.map((rr) => ({
                     id: rr.id,
-                    revisionFormData: rr.name,
+                    createdAt: rr.createdAt,
+                    updatedAt: rr.updatedAt,
+                    revisionFormData: rateFormDataToDomainModel(rr),
+
                     contractRevisions: rr.contractRevisions.map(
                         ({ contractRevision }) => ({
                             id: contractRevision.id,
                             createdAt: contractRevision.createdAt,
                             updatedAt: contractRevision.updatedAt,
                             formData:
                                 contractFormDataToDomainModel(contractRevision),
-                            rateRevisions: contractRevision.rateRevisions.map(
-                                (rr) => ({
-                                    id: rr.rateRevisionID,
-                                    revisionFormData: rr.rateRevision.name,
-                                })
-                            ),
+                            rateRevisions: [],
```

Comment:
Is `rateRevisions` still needed here? It could be confusing that these `contractRevisions` have an empty array of  `rateRevisions`. This empty string could also mean no relationships in the DB. I rather we are explicit and not include it in `finalRate`.